ha/core/event_analyzer/filter/filter.py

```diff
@@ -86,3 +102,54 @@ def filter_event(self, msg: str) -> bool:
         except Exception as e:
             raise EventAnalyzerError(f"Failed to filter event. Message: {msg}, Error: {e}")
 
+
+class IEMFilter(Filter):
+    """ Filter IEM consumed by watcher """
+
+    def __init__(self):
+        """
+        Init method
+        """
+        super().__init__()
+
+        # Get filter type and resource types list from the IEM rule file
+        self.filter_type = Conf.get(const.ALERT_FILTER_INDEX, "iem.filter_type")
+        self.components_list = Conf.get(const.ALERT_FILTER_INDEX, "iem.components")
+        self.modules_dict = Conf.get(const.ALERT_FILTER_INDEX, "iem.modules")
+
+    def filter_event(self, msg: str) -> bool:
+        """
+        Filter event.
+        Args:
+            msg (str): Msg
+        """
+        try:
+            iem_required = False
+            message = json.loads(msg)
+
+            _actuator_response_type = message.get('actuator_response_type')
+            if _actuator_response_type is not None:
+                return iem_required
+
+            _msg_type = Filter.get_msg_type(message)
+            if _msg_type.lower() != MESSAGETYPE.IEM.value.lower():
+                return iem_required
+
+            _sensor_response_type = message.get('sensor_response_type')
+            _component_type = _sensor_response_type['specific_info']['component']
+            _module_type = _sensor_response_type['specific_info']['module']
+
+            if self.filter_type == const.INCLUSION:
+                if _component_type in self.components_list and _module_type in self.modules_dict.get(_component_type):
+                    iem_required = True
+            elif self.filter_type == const.EXCLUSION:
+                if _component_type not in self.components_list or _module_type not in self.modules_dict.get(
+                        _component_type):
+                    iem_required = True
+            else:
```

Comment:
`Log.error("Invalid IEM filter type in the event IEM filter rules")`

This case should not exist 
it is validation check
can you check it in validate() function and call validate_rule in init of this class

Reply:
Done.